spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/endpoint/EndpointAutoConfiguration.java

```diff
@@ -16,30 +16,45 @@
 
 package org.springframework.boot.actuate.autoconfigure.endpoint;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.ListableBeanFactory;
+import org.springframework.beans.factory.annotation.BeanFactoryAnnotationUtils;
 import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
 import org.springframework.boot.actuate.endpoint.invoke.ParameterValueMapper;
 import org.springframework.boot.actuate.endpoint.invoke.convert.ConversionServiceParameterValueMapper;
 import org.springframework.boot.actuate.endpoint.invoker.cache.CachingOperationInvokerAdvisor;
 import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.convert.ApplicationConversionService;
+import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.convert.converter.GenericConverter;
 import org.springframework.core.env.Environment;
 
 /**
  * {@link EnableAutoConfiguration Auto-configuration} for {@link Endpoint} support.
  *
  * @author Phillip Webb
  * @author Stephane Nicoll
+ * @author Chao Chang
  * @since 2.0.0
  */
 @Configuration(proxyBeanMethods = false)
 public class EndpointAutoConfiguration {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public ParameterValueMapper endpointOperationParameterMapper() {
-		return new ConversionServiceParameterValueMapper();
+	public ParameterValueMapper endpointOperationParameterMapper(
+			ApplicationContext applicationContext) {
+		return new ConversionServiceParameterValueMapper(
+				new Factory(applicationContext.getAutowireCapableBeanFactory()).create());
```

Comment:
You are able to inject `Converter` and `GenericConverter` classes via `ObjectProvider`.
```java
	@Bean
	@ConditionalOnMissingBean
	public ParameterValueMapper endpointOperationParameterMapper(
			@EndpointConverter ObjectProvider<Converter<?, ?>> converters,
			@EndpointConverter ObjectProvider<GenericConverter> genericConverters) { 
		//
	}
```

Reply:
Thanks, @nosan. I suspect the proposal is as it is as it has been inspired by `@ConfigurationPropertiesBinding` and `ConversionServiceDeducer`. The context here is a little bit different, however, as the configuration properties stuff is used from a bean post-processor. We can take a look at make some of the amendments suggested here as part of merging this.